File: cpp/GoaBusinessSchool.cpp
#include "GoaBusinessSchool.h"
#include <iostream>

GoaBusinessSchool::GoaBusinessSchool(int id, const std::string& name) : id(id), name(name) {}

void GoaBusinessSchool::offerProgram(const Program& program) const {
    std::cout << name << " offers " << program.name << std::endl;
}

File: csharp/GoaBusinessSchool.cs
using System;

namespace StudentManagementSystem
{
    public class GoaBusinessSchool
    {
        public int Id { get; set; }
        public string Name { get; set; }

        public GoaBusinessSchool(int id, string name)
        {
            Id = id;
            Name = name;
        }

        public void OfferProgram(Program program)
        {
            Console.WriteLine($"{Name} offers {program.Name}");
        }
    }
}
